feat(apple): Add docs on file I/O APM (#4584)
getsentry/sentry-docs: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx

[NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
[NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection

## File I/O Instrumentation

<Alert level="info" title="Important">

> This is an experimental feature. Use it with caution.

Reviewer: ```suggestion
This is an experimental feature and requires an opt-in.
```

</Alert>

> The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other Apple APIs like [NSFileManager][NSFileManager] uses [NSData][NSData].

Reviewer 2: ```suggestion
The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].
```

> The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other Apple APIs like [NSFileManager][NSFileManager] uses [NSData][NSData].

Reviewer 2: `m`: Can we list the Apple APIs we know that use NSData internally, please?

Author: Well, there is a ton of APIs that use NSData internally. 
But if you mean APIs available to load data directly I could think of NSFileManager, NSString and NSBundle.
I would add this as examples, but I don't believe a complete list is relevant here.

> The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].

Reviewer 3: I know it's not consistent across the board, but mostly we don't capitalize "Scope" in this context in the docs. 

```suggestion
The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].
```

This feature is experimental and is disable by default. 

> To enable File I/O instrumentation:

Reviewer 2: ```suggestion
To enable file I/O instrumentation:
```
